src/components/Restoration/CorrectNameOptions.vue

```diff
@@ -81,17 +98,22 @@ import CorrectNameRequest from '@/components/Restoration/CorrectNameRequest.vue'
   }
 })
 export default class CorrectNameOptions extends Vue {
-  /** The options to display */
-  @Prop() readonly correctionNameChoices!: Array<string>
+  @Prop({ default: null }) readonly actionTxt!: string
+  @Prop({ default: () => [] }) readonly correctionNameChoices!: Array<NameChangeOptions>
+  @Prop({ required: true }) readonly businessId!: string
+  @Prop({ required: true }) readonly entityType!: CorpTypeCd
+  @Prop({ required: true }) readonly nameRequest!: NameRequestIF
+  @Prop({ required: true }) readonly companyName!: string
+  @Prop({ required: true }) readonly fetchAndValidateNr!: () => Promise<NameRequestIF>
```

Comment:
Look, I'm passing in a function that fetches and validates the NR (so that THIS component doesn't need to know about axios or urls, etc).

Reply:
I love the way that JavaScript makes functions first class citizens -- this is such a powerful technique!  Do we plan to pass in different functions, if so, can you provide some examples?

Reply:
This component is now in BCRS shared repo.